great_expectations/rule_based_profiler/parameter_builder/metric_multi_batch_parameter_builder.py

```diff
@@ -33,6 +33,7 @@ def __init__(
         metric_name: str,
         metric_domain_kwargs: Optional[Union[str, dict]] = None,
         metric_value_kwargs: Optional[Union[str, dict]] = None,
+        single_batch_mode: Union[str, bool] = False,
```

Comment:
Since this is new, can we make it a `bool` instead of a `Union`?

Reply:
@billdirks this is for `$variable` syntax -- we do that throughout.  Thanks!

Reply:
I see, so it is being used as input from a text file?

Reply:
@billdirks This fetches from a dictionary -- please see `great_expectations/expectations/core/expect_table_row_count_to_be_between.py` for the example (and/or `Rule` definitions in a `DataAssistant`).  Thanks!